src/cc2olx/olx.py

```diff
@@ -155,6 +163,72 @@ def _create_olx_nodes(self, content_type, details):
 
         return nodes
 
+    def _create_video_node(self, details):
+        """
+        This function creates Video OLX nodes.
+
+        Args:
+            details (Dict[str, str]): Dictionary that has Video tag value.
+
+        Returns:
+            [OLX Element]: Video OLX element.
+        """
+        xml_element = element_builder(self.doc)
+        attributes = {
+            "youtube": "1.00:" + details["youtube"],
+            "youtube_id_1_0": details["youtube"]
+        }
+        child = xml_element("video", children=None, attributes=attributes)
+        return child
+
+    def _process_html(self, details):
+        """
+        This function helps to process the hlml and gives out
+        corresponding HTML or Video OLX nodes.
+
+        Args:
+            details (Dict[str, str]): Dictionary that has HTML tag value.
+
+        Returns:
+            List[OLX Element]: List of html/Video OLX element.
+        """
+        video_olx = []
+        nodes = []
+        child = self.doc.createElement("html")
+        html = self._process_static_links(details["html"])
+        if self.link_file:
+            html, video_olx = self._process_html_for_iframe(html)
+        txt = self.doc.createCDATASection(html)
+        child.appendChild(txt)
+        nodes.append(child)
+        for olx in video_olx:
+            nodes.append(olx)
+        return nodes
+
+    def _process_html_for_iframe(self, html_str):
+        """
+        This function helps to parse the iframe with
+        embedded video, to be converted into video xblock.
+
+        Args:
+            html_str ([str]): Html file content.
+
+        Returns:
+            html_str [str]: The html content of the file, if iframe is present
+                            and converted into xblock then it is clipped before
+                            returning.
+            video_olx [List[xml]]: List of xml children, i.e video xblock.
+        """
+        video_olx = []
+        parsed_html = html.fromstring(html_str)
+        iframes = parsed_html.xpath("//iframe")
+        if not iframes:
+            return html_str, video_olx
+        video_olx = self.iframe_link_parser.get_video_olx(self.doc, iframes)
+        if video_olx:
+            return html.tostring(parsed_html).decode('utf-8'), video_olx
```

Comment:
>  Is it because get_video_olx has the potential to modify the HTML (i.e. remove iframes), so you need a new string representation of the changed HTML?

Yes you are right, and this will only happen if it has a video xblock generated.

Hence we need to serve the HTML where iframe is removed.